### ISSUE TITLE
Automation Proposal

### ISSUE THREAD

**Issue by 100AutomationsProject** [100Automations/futureautomations#49]
## Your Name
Ava Li

## Your Email
avali4907@outlook.com

## Overview: describe the kind of repetitive thing you have to do and the benefit of automating it?
Not repetitive per se, but there is a lot of project management that can be simplified through storypoints. This project aims to add storypoints to the GitHub board via browser extension for Firefox (and Chrome in the future)

## Ownership of Idea
This is my idea and I need help developing it

## Current State: Is this just an idea, or has any work already been done to build it?
Work has been done here, as described here: https://github.com/hackforla/ops/issues/18#issuecomment-1060875940. I will need to add a few things here and there to make the extension more robust and customizable, but overall the code referenced there will serve as the base.

## Languages or frameworks used?
Typescript

## Platform?
Firefox

## Automation triggers: What starts it? What's it responding to? Time or event based?
None. It can be switched on or off and it will continue to work. Rather than automate the task of counting issues by hand, it elliminates the task altogether by using data and programming to count.

## How much manual or custom input is required?
Very little.

## Output: What's the desired result? What do we not want to see?
Displaying some widget in the Github Project Board with story points.

## Project size: rough estimate of effort and time needed to build it
Probably less than 2000 lines of code, so with an experienced coder, it should take maybe a week?

## Stakeholders and Impact - who benefits and how?
Anyone who uses storypoints to manage issues.

## Resources/Instructions: Has any documentation been created yet?
Nope.

## Action Items/Research Needed
Unsure.

**Comment by Aveline-art** [comment]
Update: after a three day sprint, I am almost done. Repo, [here](https://github.com/Aveline-art/foxy-pangolins). The only blockers are

1) completing the ui to set a custom regex
2) completing storage to store ui state
3) submitting it to the firefox board for review

That said, is there a process for getting a designer to help with UI design? I am unable to create a "pretty" ui on my own.

**Comment by Aveline-art** [comment]
Update: The functionality is pretty much all done. The only blockers are:

1. implementing a user feedback alert
2. submitting it to the firefox board for review

**Comment by ExperimentsInHonesty** [comment]
@Aveline-art Yeah!! Another fantastic automation!  Since you are a return contributor, I am going to consolidate the rest of our questions in one place

### Items for us to address:
- 1. Do you have a name you have picked for the automation?  Please see [Name your Automation](https://100automations.org/guides/name-your-automation.html) for guidelines.  We can pick one for you if you need us to.
- 2. Do you need us to create an image for it?  Have any ideas to share?
- 3. Are you ready for us to import the repo to 100automations github org?
- 4. Don't forget to work on the health files: https://100automations.org/guides/community-support-for-automations.html
- 5. [Write a Short Narrative Description about Your Automation](https://100automations.org/guides/writing-short-descriptions-for-automations.html) to go on the project card
6. Status
#### Status
Status of project: (must pick one)
- [ ] In Use And Ready To Deploy
- [ ] In Use and Needs Further Development
- [ ] In Use And Being Refactored
- [ ] Ready To Deploy
- [ ] In Development
- [ ] Ready For Development
- [ ] In Ideation
- [ ] Abandoned

Needs Help status: (can pick as many as apply)
- [ ] Needs Maintenance
- [ ] Needs Support For Users
- [ ] Needs Refactoring
- [ ] Needs Documentation
- [ ] Needs Peer Review
- [ ] Needs To Be Built
- [ ] Needs To Be Researched
- [ ] Needs Advice
- [ ] Needs Takeover
- [ ] Advice Needed

**Comment by ExperimentsInHonesty** [comment]
also, would like to know if you anticipate writing the chrome extension after you get it working in Firefox?  If not, let us know so we can recruit someone to do the conversion.

BTW here is a link on the topic https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Build_a_cross_browser_extension

**Comment by Aveline-art** [comment]
>Do you have a name you have picked for the automation? Please see [Name your Automation](https://100automations.org/guides/name-your-automation.html) for guidelines. We can pick one for you if you need us to.

So far it is called foxy-pangolins, but since we are staying away from codenames, I would love suggestions on some kind of internal brand name (kind of like how we have FB's React, or Ruby's Unicorn), just to simplify things for me.

An AI generator gave me GitHub-Spoints, which I find endearing. Thoughts?

> Do you need us to create an image for it? Have any ideas to share?

Yes, please! I am not much of a graphics person.

> Are you ready for us to import the repo to 100automations github org?

Just create a link to the repo and give me permission to write into it and I can handle the rest.

> Don't forget to work on the health files: https://100automations.org/guides/community-support-for-automations.html
[Write a Short Narrative Description about Your Automation](https://100automations.org/guides/writing-short-descriptions-for-automations.html) to go on the project card

I assume some number of these will come with the repo?

**Comment by henlatourrette** [comment]
Hi @Aveline-art , we've cloned your repo and rebranded it to "[Github SPoints Counter](https://github.com/100Automations/github-spoints-counter)" . Hopefully, that name is fine for you as well :) 

We've also added you to the "write" team for that repo as a maintainer, can you please check whether your permissions are working fine and confirm back? 

Regarding your question about the `docs/ `directory, I'm fine with that as long as it is explained in the Readme file or somewhere under the installation instructions. 
As for the health files, those are the basic files required in order to get support from the community, you can follow [this guide](https://100automations.org/guides/community-support-for-automations.html) to help you with that.

As for the UX part, we don't have an established process for that yet but we could work together on that while we move forward with the other steps. 
Let us know if you need any help or have more comments on the process!

**Comment by Aveline-art** [comment]
Hi @henlatourrette 

Thank you for setting all this up and getting back to me. Everything looks great!

Just one small tidbit. You might have added me to the wrong repository as I do not seem to have permission (see image below).

![no permissions](https://user-images.githubusercontent.com/62368440/163925028-42434281-f10c-4e06-aa92-cae8f53f2b79.png)

**Comment by henlatourrette** [comment]
> Hi @henlatourrette
> 
> Thank you for setting all this up and getting back to me. Everything looks great!
> 
> Just one small tidbit. You might have added me to the wrong repository as I do not seem to have permission (see image below).
> 
> ![no permissions](https://user-images.githubusercontent.com/62368440/163925028-42434281-f10c-4e06-aa92-cae8f53f2b79.png)

Hi @Aveline-art, thanks for your feedback. I'm reviewing the teams and permissions now and will let you know when it's solved.

**Comment by henlatourrette** [comment]
@Aveline-art The permissions issue should be solved by now - I've made some updates on the "Team" configuration. Please try again and let me know if you have any other issues with that. Thanks!

In parallel, we'll try to come up with an idea for the logo image.

**Comment by Aveline-art** [comment]
`@henlatourrette @ExperimentsInHonesty `

I have been reading through the FireFox extension publishing steps (there are a lot). These are some of the steps I need to do before submitting:

- Follow trademark policy. As we use GitHub as part of its name, we need to make sure we comply with [GH's trademark policy](https://docs.github.com/en/site-policy/content-removal-policies/github-trademark-policy). Do we have any legal person who can double check and weigh in on this?

**Comment by henlatourrette** [comment]
> 

Hi @Aveline-art , yes we agree on using the self-publish option and we would be happy of testing on the 100automations website repo and we can recruit some other teams to test it if you want.

How many people would you need to test it?

**Comment by Aveline-art** [comment]
Hi @henlatourrette . Sorry it took me a week to get back to you. I have finally managed to self-publish the add-on. It seems to distribute it via self-publish, I would need to add their firefox developer account to the extension to allow them to download it.

Since this is such a hassle to do for now, I think it would be best if I can get a volunteer for some initial feedback to make sure the functionality is on the right track?

**Comment by Aveline-art** [comment]
@henlatourrette @ExperimentsInHonesty 

I have updated [README ](https://github.com/100Automations/github-spoints-counter#readme)for the automation with new instructions. You should no longer need a developer account. Hopefully this will be easier to use and test!

**Comment by Aveline-art** [comment]
Hi! Just a follow up to the meeting today:

In addition to the look and feel of the popup, something else I would need feedback on are the [instructions on how to use it](https://100automations.github.io/github-spoints-counter/#steps). When an extension is first downloaded, what is supposed to happen is that a new tab pops up to a website with instructions on how to use the extension. I know the design is currently very lack-luster, but I want to make sure that the instructions are clear and understandable.

**Comment by henlatourrette** [comment]
@Aveline-art we are changing the name to "GitHub Story Points Calculator" -  we also added an issue on its repo.
